contributor_docs/creating_libraries.md

```diff
@@ -282,6 +282,31 @@ if (typeof p5 !== undefined) {
 
 In the above snippet, an additional `if` condition is added around the call to `p5.registerAddon()`. This is done to support both direct usage in ESM modules (where users can directly import your addon function then call `p5.registerAddon()` themselves) and after bundling support regular `<script>` tag usage without your users needing to call `p5.registerAddon()` directly as long as they have included the addon `<script>` tag after the `<script>` tag including p5.js itself.
 
+## Accessing custom actions
+In certain circumstances, such as when you have a library that listens to a certain browser event, you may wish to run a function that your user defined on the global scope, much like how a `click` event triggers a user defined `mouseClicked()` function. We call these functions "custom actions" and your addon can access any of them through `this._customActions` object.
+
+The following addon snippet listens to the `click` event on a custom button element.
+```js
+function myAddon(p5, fn, lifecycles){
+  lifecycles.presetup = function(){
+    let customButton = this.createButton('click me');
+    customButton.elt.addEventListener('click', this._customActions.myAddonButtonClicked);
```

Comment:
Minor: Include very brief description of what will happen if "myAddonButtonClicked" function does not exist?

Reply:
If `myAddonButtonClicked` does not exist, the value will return `undefined` and in the context of `addEventListener` will be essentially a no op. Would it need to be documented as so since I think it may be relatively intuitive.

Reply:
Yes that's a good point to mention, I'll add a line about it.